.gitignore

```diff
@@ -1,2 +1,10 @@
 .DS_Store
-.idea
\ No newline at end of file
+.idea
+out.txt
+1.txt
+test.txt
+test1.txt
+test2.txt
+coverage.out
+bin/
+tests/
```

Comment:
No need to ignore `tests` directory because there can be real directory with tests. Create another named directory for test purpose, e.g. `.data`, `tmp` etc